src/nni_engine.hpp

```diff
@@ -60,48 +68,141 @@ class NNIEngine {
   void RunMainLoop();
   void RunPostLoop();
 
-  // ** Evaluation/Ranking
-  // These evaluate all NNIs from the NNISet by a criterion in order to place
-  // them in a relative ordering.
-
-  // Evaluate all adjacent NNIs.
-  void EvaluateAdjacentNNIs();
-  // Method Selector for Individual NNI Evaluation.
-  double EvaluateNNI(const NNIOperation &proposed_nni);
-  // Compute NNI Likelihood, via GPEngine's quartet request.
-  double ComputeNNILikelihood(const NNIOperation &nni);
-  // Add score to given NNI.
-  void AddScoreForNNI(const NNIOperation &nni, const double score);
-  double GetScoreForNNI(const NNIOperation &nni) const;
+  // ** Filter Functions
 
-  // ** Filtering
+  // Initialization step for filter before beginning
+  using StaticFilterInitFunction =
+      std::function<void(NNIEngine &, GPEngine &, GraftDAG &)>;
```

Comment:
This seems great to me, though I want to double check with @ognian- who has some opinions regarding passing functions. We can discuss tomorrow.